spec/models/post_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  it 'is valid with valid attributes' do
+    user = User.create(name: 'John Doe')
+    post = Post.new(author: user, title: 'My Post', text: 'This is my post content.')
+    expect(post).to be_valid
+  end
+
+  it 'is not valid without a title' do
+    post = Post.new(title: nil)
+    expect(post).not_to be_valid
+  end
+
+  it 'is not valid with a blank title' do
+    post = Post.new(title: '')
+    expect(post).not_to be_valid
+  end
+
+  it 'is not valid with a title exceeding 250 characters' do
+    long_title = 'a' * 251
+    post = Post.new(title: long_title)
+    expect(post).not_to be_valid
```

Comment:
- [OPTIONAL ] Great job for adding the tests for the post validation but I think it would be great if you add tests for the methods here as mentioned in the requirements to test all the methods so please test increment_user_posts_counter and recent_comments that in your post model 🔥⭐♻ I am assigning it as optional because the previous reviewer didn't mention it but it still important to do it 🔥


![image](https://github.com/ReemMohamedAbdelfatah/rails-blog-app/assets/58553711/9fd3c9d3-a3e4-4ad2-b7c2-48166d868e87)

Reply:
Hello Reem It is already done if you switched to the test-and-validations branch. Thank you for your time.